irc/websocket.go

```diff
@@ -0,0 +1,56 @@
+package irc
+
+import (
+	"github.com/gorilla/websocket"
+	"net"
+	"net/http"
+	"time"
+)
+
+var upgrader = websocket.Upgrader{
+	ReadBufferSize:  1024,
+	WriteBufferSize: 1024,
+	CheckOrigin:     func(r *http.Request) bool { return true },
```

Comment:
If you dont accept from any origin, the server will not accept connection from client, by "conception" they are not on the same origin.
The js code (the client), is not hosted by the IRC server but by the webpage who wants to add a chat system.